### ISSUE TITLE
2.1 — Recover the trigger recall lost to the rename

### ISSUE THREAD

**Issue by urmindra** [realus99/dopod-design#4]
Renaming `carbon-design` to `dopod-design` cost measured trigger accuracy: **18/20 to 16/20, recall 8/10 to 6/10**, zero false positives throughout. The old name was itself a signal to the model; the new one is opaque, so the description now carries all the weight.

A front-loaded description rewrite leading with "IBM Carbon Design System (v11)" was tested and did **not** recover it — same 16/20, same four misses. The lever is not wording.

**Options, cheapest first:**

1. **Re-run the description optimizer under the current name.** The previous run optimized against `carbon-design`, so its winner is tuned for a signal that no longer exists. Untried and most likely to pay off. Use the patched harness (see #7.1, #7.2), not the stock one.
2. Test whether the opening lines of the `SKILL.md` **body** affect loading, not just the frontmatter description.
3. Accept 16/20 and document it as the cost of the branding decision.

All four misses are prompts where Carbon is the house style but the word never appears — the exact case the skill exists to catch.

## Success criteria

_Observable outcomes. How we know this worked._

- [ ] Trigger rate re-measured on the full 20-query set at 3 runs per query using the patched harness
- [ ] Either recall reaches ≥8/10, or a written finding documents the ceiling and lists every variant tried
- [ ] Precision does not regress — false positives stay at 0

## Definition of done

_The completeness bar for closing this issue._

- [ ] Measurement artifacts committed so the numbers are reproducible, not asserted
- [ ] `SKILL.md` description updated if a winner emerges
- [ ] `CHANGELOG.md` records before/after numbers
- [ ] If the ceiling is confirmed, README known-limitations says so plainly
- [ ] `npm test` passes; new behaviour has tests covering it
- [ ] `npm run build` succeeds and is reproducible (a rebuild yields identical file hashes)
- [ ] No new runtime dependencies — the package stays zero-dependency
- [ ] `CHANGELOG.md` updated under the target version
- [ ] User-visible changes reflected in `README.md` and `--help` where relevant
- [ ] Merged to `main` via PR; this issue closed by that merge

---
From the v1.0.0 backlog. Context: https://github.com/realus99/dopod-design/blob/main/docs/BACKLOG.md

**Comment by urmindra** [comment]
**Blocked on #24.** Started this and found the measurement itself is unreliable.

The harness counts a trigger only when its planted proxy is invoked, but `~/.claude/skills/dopod-design-carbon` is installed globally and is a near-identical sibling skill. Traced on one query, the model invoked both:

```
skills invoked: ["dopod-design-carbon", "dopod-design-skill-probeaaa"]
```

Which one it reaches for is stochastic — identical 6-query runs scored 3/6, 4/6, 5/6, 4/6, with the same query returning rate 1.0 and 0.0 on different attempts.

Two contributing bugs fixed along the way (probe now runs from a neutral dir outside the repo; harness no longer gives up on the first non-matching tool call — #25), but variance remains high because the model is genuinely choosing between two skills that do nearly the same thing.

A 40-query baseline run produced 21/40 with recall 1/20, which is **not a real result** — it predates both fixes and is contaminated. Disregard it.

Do #24 first, then re-measure on `trigger-eval-v2.json`.

**Comment by urmindra** [comment]
**Unblocked** — #24 is closed and the harness is now stable (three identical runs, exact agreement).

**New baseline, shipped description, 40-query v2 set, isolated:**

| | |
|---|---|
| Correct | 31/40 |
| Recall | 13/20 |
| Precision | 13/15 |
| False positives | 2 |

This supersedes every earlier figure. Recall is 13/20, not 6/10 — the contaminated runs were blaming the description for the model choosing a sibling skill.

**Misses (all implicit positives):** ops console, confirm-before-delete, labels washing out on dark themes, panel loading state, pagination, chart colours clashing with status badges, build-from-mockup.

**False positives:** tree-shaking `@carbon/react`, migrating away from Carbon.

Both directions now have headroom, so the optimizer has something to work with. Next step is a full run against `trigger-eval-v2.json` using the patched harness — roughly 40-50 min for 5 iterations at 40 queries × 3 runs.

Note for whoever picks this up: the rename-cost claim in the package CHANGELOG ("18/20 → 16/20") predates isolation and should be re-measured or softened.

**Comment by urmindra** [comment]
**Complete — closing on the "document the ceiling" branch of the success criteria.**

Ran 5 optimizer iterations against the isolated 40-query v2 set (24 train / 16 test, 3 runs per query, ~35 min).

## Result: no candidate beat the shipped description

| Iteration | Train | Test | Test precision | Test recall |
|---|---|---|---|---|
| 1 (shipped) | 20/24 | **10/16** | fp=2 | 4/8 |
| 2 | 21/24 | 8/16 | fp=2 | 2/8 |
| 3 | 20/24 | 10/16 | fp=2 | 4/8 |
| 4 | 21/24 | 10/16 | **fp=1** | 3/8 |
| 5 | 21/24 | 9/16 | fp=2 | 3/8 |

Best test score 10/16, tied by iterations 1, 3 and 4 — ties go to the incumbent. Iteration 2 is a clean demonstration that the holdout now works: it improved train and regressed test, i.e. overfitting the optimizer could not previously detect.

## The ceiling, measured

Of 40 queries across 5 iterations: 26 always correct, 7 unstable, **7 always failed regardless of wording**.

Six of those seven are misses sharing one shape — an ordinary UI task with zero design-system vocabulary. Full analysis in the new issue below.

**Conclusion: description wording is exhausted.** Every variant the optimizer produced, plus the two I hand-wrote earlier, land in the same band. Remaining recall needs a different mechanism, filed as #26 — an always-on layer for Claude Code, which every other target already has.

## Variants tried

Original, optimizer iterations 2-5, a hand-written front-loaded variant, and the shipped wording — all measured on the same set under isolation.

The public CHANGELOG claim ("16/20", "rename cost 2 queries of recall") has been corrected: both were measured pre-isolation and were wrong.
